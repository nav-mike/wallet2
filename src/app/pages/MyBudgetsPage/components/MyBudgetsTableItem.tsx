import React from 'react';
import { IBudget } from './MyBudgetsTable';

export const MyBudgetsTableItem = (props: { item: IBudget }) => {
  return (
    <tr key={`${props.item.month}-${props.item.year}`}>
      <td>{props.item.year}</td>
      <td>{props.item.month}</td>
      <td>{props.item.value}</td>
    </tr>
  );
};
