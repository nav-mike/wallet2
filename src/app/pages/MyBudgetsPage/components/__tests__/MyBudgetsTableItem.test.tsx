import { render } from '@testing-library/react';
import * as React from 'react';
import { MyBudgetsTableItem } from '../MyBudgetsTableItem';

describe('<MyBudgetsTableItem />', () => {
  const data = { month: 'January', year: 2012, value: 100 };

  it('should render and match the snapshot', () => {
    const item = render(
      <table>
        <tbody>
          <MyBudgetsTableItem item={data} />
        </tbody>
      </table>,
    );
    expect(item.container.firstChild).toMatchSnapshot();
  });
});
