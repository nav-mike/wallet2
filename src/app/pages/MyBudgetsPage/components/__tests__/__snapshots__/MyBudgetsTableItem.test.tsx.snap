// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyBudgetsTableItem /> should render and match the snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        2012
      </td>
      <td>
        January
      </td>
      <td>
        100
      </td>
    </tr>
  </tbody>
</table>
`;
